packages/x-date-pickers/src/internals/models/props/baseToolbarProps.ts

```diff
@@ -8,7 +8,7 @@ export interface BaseToolbarProps<TDate>
   extends ExportedCalendarPickerProps<TDate>,
     ExportedClockPickerProps<TDate> {
   ampmInClock?: boolean;
-  date: TDate;
+  date: TDate | null;
```

Comment:
There is nothing preventing the date from being null, it is even the default value in most of our demos